modules/nixos/shared-darwin.nix

```
@@ -0,0 +1,8 @@
+{ config, lib, ...}: {
+  launchd.daemons.nix-gc.command = lib.mkForce ''
```

Worth a comment to explain why this was added?

```suggestion
  # If the machine runs low on disk space it's possible for derivation files to be created but never get content which results in derivations that can't be removed by the gc. This is a workaround which finds and deletes those problem derivations
  launchd.daemons.nix-gc.command = lib.mkForce ''
```

---

because of the duplication we may want to copy this to the linux version too, WDYT?

---

i added it to the commit message

---

Probably worth adding it to both yes